src/bootstrap/dylib_util.rs

```diff
@@ -26,3 +26,15 @@ pub fn dylib_path() -> Vec<PathBuf> {
     };
     env::split_paths(&var).collect()
 }
+
+lazy_static::lazy_static! {
+    static ref LLD_NEWER: bool = env::var_os("RUSTC_LLD_NEWER").is_some();
```

Comment:
`env::var_os("RUSTC_LLD_NEWER")` cannot be used inside `src/bootstrap` binary, it's only set for tools that bootstrap calls, like `src/bootstrap/bin/rustdoc.rs`.

The static should be initialized from the result of `Command::new("lld")...`, then then env var should be initialized from the static.

Also the static can keep the string `"--threads=1"` itself instead of the boolean flag.